### ISSUE TITLE
Index route=road relations

### ISSUE THREAD

**Issue by naoliv** [osmandapp/Osmand#1763]
Could osmand index `type=route` + `route=road` relations?
Sometimes they are used to designated a whole segment with a different/alternative or touristic name, for example.

It would be good if we could find and use them.

**Comment by vshcherb** [comment]
Yes we have similar issue with other routes, bus routes, mtb routes. Currently we don't have a solution.

**Comment by pelderson** [comment]
I am not sure this is the same issue as #2414. The word "Index", I don't know what is meant by that, also route=road does not ring a bell. 
Hiking routes are already displayed on the map (overlay) and also live available from OSM database. Given the relation number, you can extract a complete hiking route from OSM in the form of gpx, essentially a series of waypoints. OsmAnd can navigate along a gpx. 
You would need a way to select a hiking route relation for navigation. Extracting the hiking route relation as gpx from osm is a live function, but once the gpx is stored on the device you can navigate offline. 
For selection, I like the way waymarkedtrails does that. On the map screen, tap a button Routes. You'll get the routes namelisted that pass through the visible area. Tap one and you get some details and functions. 
Functions could be "Get gpx" to download gpx for later navigation; "Navigate" to download gpx and start navigation immediately after. First version could simply be an add-on to find and download a route by name or ref (which the user can see on the map), then the user can use Navigate along gpx.
